tools/packaging/packaging.psm1

```diff
@@ -30,226 +30,261 @@ function Start-PSPackage {
+        } else {
+            New-PSOptions -Configuration "Release" -WarningAction SilentlyContinue | ForEach-Object { $_.Runtime, $_.Configuration }
+        }
+
+        if($Environment.IsWindows) {
+            # Runtime will always be win7-x64 or win7-x86 on Windows.
+            # Build the name suffix for universal win-plat packages.
+            $NameSuffix = $Runtime -replace 'win\d+', 'win'
+        }
 
-    $Script:Options = Get-PSOptions
+        log "Packaging RID: '$Runtime'; Packaging Configuration: '$Configuration'"
 
-    $crossGenCorrect = $false
-    if ($Type -eq "deb-arm") {
-        # crossgen doesn't support arm32 yet
-        $crossGenCorrect = $true
-    }
-    elseif(-not $IncludeSymbols.IsPresent -and $Script:Options.CrossGen) {
-        $crossGenCorrect = $true
-    }
-    elseif ($IncludeSymbols.IsPresent -and -not $Script:Options.CrossGen) {
-        $crossGenCorrect = $true
-    }
+        $Script:Options = Get-PSOptions
 
-    # Make sure the most recent build satisfies the package requirement
-    if (-not $Script:Options -or                                ## Start-PSBuild hasn't been executed yet
-        -not $crossGenCorrect -or                               ## Last build didn't specify '-CrossGen' correctly
-        $Script:Options.Runtime -ne $Runtime -or                ## Last build wasn't for the required RID
-        $Script:Options.Configuration -ne $Configuration -or    ## Last build was with configuration other than 'Release'
-        $Script:Options.Framework -ne "netcoreapp2.0")          ## Last build wasn't for CoreCLR
-    {
-        # It's possible that the most recent build doesn't satisfy the package requirement but
-        # an earlier build does.
-        # It's also possible that the last build actually satisfies the package requirement but
-        # then `Start-PSPackage` runs from a new PS session or `build.psm1` was reloaded.
-        #
-        # In these cases, the user will be asked to build again even though it's technically not
-        # necessary. However, we want it that way -- being very explict when generating packages.
-        # This check serves as a simple gate to ensure that the user knows what he is doing, and
-        # also ensure `Start-PSPackage` does what the user asks/expects, because once packages
-        # are generated, it'll be hard to verify if they were built from the correct content.
-        $params = @('-Clean')
-        if(-not $IncludeSymbols.IsPresent)
-        {
-            $params += '-CrossGen'
+        $crossGenCorrect = $false
+        if ($Type -eq "deb-arm") {
+            # crossgen doesn't support arm32 yet
+            $crossGenCorrect = $true
+        }
+        elseif(-not $IncludeSymbols.IsPresent -and $Script:Options.CrossGen) {
+            $crossGenCorrect = $true
+        }
+        elseif ($IncludeSymbols.IsPresent) {
```

Comment:
So now it's OK to specify `-IncludeSymbols` while build with `-Crossgen`?

Reply:
Yes, the non-publish builds always keep the symbols, unaltered.